recipes/default_java_symlink.rb

```diff
@@ -12,8 +12,24 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+require 'pathname'
 
-link '/usr/lib/jvm/default-java' do
-  to node['java']['java_home']
-  not_if { node['java']['java_home'] == '/usr/lib/jvm/default-java' }
```

Comment:
Doesnt this not if do what all the code below does anyways? Or if theres a different check we can add to the guard to do what you're trying to do below?